regression/cbmc-outfile/chain.py

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env python3
```

Comment:
Do we really need that script? Couldn't it also be done by using `--outfile /dev/stdout`?

Reply:
Yes, unfortunately we need it as:

1. We test on windows where `/dev/stdout` may not work
2. By adding `--outfile /dev/stdout` we will know that the SMT formula is printed on the screen, not that it is added to the out file, so we cannot distinguish file from normal output.
3. The way the test output is checked is not in a certain order while the python script enforce a certain ordering, so if we want to have 2 `(check-sat)` calls the old will be satisfied with one `(check-sat)` in the output, the new one instead will require 2 in the specified order

Reply:
1. I believe you can use `-` instead of `/dev/stdout`, which will then work across platforms.
2. Granted, but I'm not sure this one aspect is a sufficient reason to warrant yet another piece of code that needs to be maintained.
3. You can use `activate-multi-line-match` to handle this case.

I don't intend to block on that, but every line of code that is written has a (future) cost. So I ask for careful consideration whether this really is needed.

Reply:
Unfortunately in the new SMT backend we do not support the old `--outfile -` behaviour as, because of the interactive behaviour, the output will be interleaved with other messages (defying the sense of `--outfile`).
It is still possible to get the SMT formula on the `stdout` by using `--verbosity 10` and then filtering the lines sent to the solver.

Reply:
Hmm, good point. So I guess you have to keep the Python script, and I'm just hoping it won't break...

Reply:
> 1. I believe you can use `-` instead of `/dev/stdout`, which will then work across platforms.

The `-` option is special cased in the solver factory. So tests which rely on this aren't fully testing the code path for writing to file.

I think this python script has value beyond the new incremental smt backend, because it could also be used to test the dimacs output for example.